src/Store/SingleTableStore.php

```diff
@@ -220,4 +271,39 @@ public function schema(): Schema
 
         return $schema;
     }
+
+    private function splitEventStream(Message $message): void
+    {
+        $toBeArchivedMessages = $this->loadNotArchived(
+            $message->aggregateClass(),
+            $message->aggregateId(),
+            $message->playhead()
+        );
+
+        foreach ($toBeArchivedMessages as $toBeArchivedMessage) {
+            if ($toBeArchivedMessage->archived()) {
+                // already archived, should we exit here since there seems something wrong with the
+                // query?
+            }
+
+            $this->archiveMessage($toBeArchivedMessage);
+        }
+    }
+
+    private function archiveMessage(Message $toBeArchivedMessage): void
+    {
+        $archivedMessage = $toBeArchivedMessage->withArchived(true);
+
+        $this->connection->update(
+            $this->storeTableName,
+            [
+                'archived' => $archivedMessage->archived(),
+            ],
+            [
+                'aggregate' => $this->aggregateRootRegistry->aggregateName($archivedMessage->aggregateClass()),
+                'aggregate_id' => $archivedMessage->aggregateId(),
+                'playhead' => $archivedMessage->playhead(),
+            ]
+        );
```

Comment:
With this, we should be really careful. Since we dont want to update the stream. So maybe another solution should be taken? Like creating a new stream and let the old one in the db, like versioning.